### PR TITLE
HPCC-14826 Remove calls to removeAssociation and addGroup()

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review.  It has no affect on most examples, but significantly effects several examples (e.g., see assertmatch1.eclxml).
The jira contains comments on why the change is valid.

File: ecl/hqlcpp/hqlcppds.cpp [hunk 1]
```diff
@@ -4797,7 +4790,6 @@ IHqlExpression * HqlCppTranslator::ensureIteratedRowIsLive(BuildCtx & initctx, B
     }
 
     BuildCtx childctx(iterctx);
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlcppds.cpp]: Is childctx still needed or can iterctx be used instead?

**Comment by jamienoss** [comment]
@ghalliday only one minor comment.

File: ecl/hqlcpp/hqlcppds.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlcppds.cpp]: It could probably be removed, but I played it safe on this particular case.

**Comment by ghalliday** [comment]
@richardkchapman

**Pull request merged**